variables/builtin.go

```diff
@@ -70,6 +71,15 @@ func SetPlatformArgs(s *Scope, platform specs.Platform) {
 	s.AddInactive("TARGETVARIANT", platform.Variant)
 }
 
+// SetUserPlatformArgs sets the user's platform-specific built-in args.
+func SetUserPlatformArgs(s *Scope) {
+	platform := platforms.DefaultSpec()
+	s.AddInactive("USERPLATFORM", platforms.Format(platform))
+	s.AddInactive("USEROS", platform.OS)
+	s.AddInactive("USERARCH", platform.Architecture)
+	s.AddInactive("USERVARIANT", platform.Variant)
+}
```

Comment:
Can you also add these to targetinput.go?

Reply:
I think I already did. It's [here](https://github.com/earthly/earthly/blob/9f5eddafd447182dc8c950f7a1bc972df311a9a6/states/dedup/targetinput.go#L164-L167), right?

Reply:
Ah yes - indeed